scheduling/storage_provider_tests.go

```diff
@@ -0,0 +1,69 @@
+package scheduling
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestMemoryStorageProviderSet(t *testing.T) {
```

Comment:
Lets make these tests generic so we can use the same ones against the ETCD provider later

Reply:
I.E. Set these all up as a suite and pass the provider into the tests